CONTRIBUTING.md

```diff
@@ -0,0 +1,74 @@
+
+## Contributor License Agreements
+
+We'd love to accept your patches! Before we can take them, we have to jump a
+couple of legal hurdles.
+
+The [Microsoft CLA](https://cla.microsoft.com/) must be signed by all contributors. Please fill out either the individual or corporate Contributor License Agreement (CLA). Once you are CLA'ed, we'll be able to accept your pull requests. Once you are CLA'ed, we'll be able to accept your pull requests.
+
+***NOTE***: Only original source code from you and other people that have
+signed the CLA can be accepted into the repository.
+
+
+## Support Channels
+
+Whether you are a user or contributor, official support channels include:
+
+GitHub issues:
+- ACS-Engine: https://github.com/Azure/acs-engine/issues - file issues and PRs related to ACS-Engine
+- ACS: https://github.com/Azure/acs/issues - file issues and PRs related to Azure Container Service
+
+Before opening a new issue or submitting a new pull request, it's helpful to search the project - it's likely that another user has already reported the issue you're facing, or it's a known issue that we're already aware of.
+
+## Milestones
+We use milestones to track progress of releases. There are also 2 special milestones
+used for helping us keep work organized: `Upcoming - Minor` and `Upcoming - Major`
+
+`Upcoming - Minor` is used for keeping track of issues that aren't assigned to a specific
+release but could easily be addressed in a minor release. `Upcoming - Major` keeps track
+of issues that will need to be addressed in a major release. For example, if the current
+version is `2.2.0` an issue/PR could fall in to one of 4 different active milestones:
+`2.2.1`, `2.3.0`, `Upcoming - Minor`, or `Upcoming - Major`. If an issue pertains to a
+specific upcoming bug or minor release, it would go into `2.2.1` or `2.3.0`. If the issue/PR
+does not have a specific milestone yet, but it is likely that it will land in a `2.X` release,
+it should go into `Upcoming - Minor`. If the issue/PR is a large functionality add or change
+and/or it breaks compatibility, then it should be added to the `Upcoming - Major` milestone.
+An issue that we are not sure we will be doing will not be added to any milestone.
+
+A milestone (and hence release) is considered done when all outstanding issues/PRs have been closed or moved to another milestone.
+
+## Issues
+Issues are used as the primary method for tracking anything to do with the acs-engine project.
+
+### Issue Lifecycle
+The issue lifecycle is mainly driven by the core maintainers, but is good information for those
+contributing to acs-engine. All issue types follow the same general lifecycle. Differences are noted below.
+1. Issue creation
+2. Triage
+    - The maintainer in charge of triaging will apply the proper labels for the issue. This
+    includes labels for priority, type, and metadata (such as "orchestrator/k8s"). The only issue
```

Comment:
Currently our priority has been the following:
1. P0 - means we should fix in the next few weeks
2. P1 - means we should fix in the next few months
3. P2 - let's leave it open to see what interest this gains, but if it sits dormant we close after 3-6 months.